Source/cmGlobalVisualStudio8Generator.cxx

```diff
@@ -227,9 +227,14 @@ void cmGlobalVisualStudio8Generator::WriteSolutionConfigurations(
   std::ostream& fout, std::vector<std::string> const& configs)
 {
   fout << "\tGlobalSection(SolutionConfigurationPlatforms) = preSolution\n";
-  for (std::string const& i : configs) {
-    fout << "\t\t" << i << "|" << this->GetPlatformName() << " = " << i << "|"
-         << this->GetPlatformName() << "\n";
+   std::vector<std::string> platforms = { this->GetPlatformName() };
+  if (this->GetBuildAsX())
+    platforms.push_back("ARM64");
```

Comment:
should this be ARM64X?

Also nit: add braces around statement body (although I'm not 100% sure if that's part of the official CMake style guidelines)

Reply:
This should be arm64